packages/coverage-istanbul/src/provider.ts

```diff
@@ -386,6 +386,11 @@ export class IstanbulCoverageProvider
       resolve(this.ctx.config.root, file),
     )
 
+    const viteNodeServers = [
+      this.ctx.vitenode,
+      ...this.ctx.projects.map(project => project.vitenode),
```

Comment:
Did you consider filtering servers by root first?

Reply:
I had something like that initially: https://github.com/AriPerkkio/vitest/commit/832d0b4a690376bc2d52b796c24b4d2f87409ae0#diff-a0a730c31ce78cf1c3cb20d55daa7167ff57fa8e062df213c6eeb918582d1cb5R282-R303

But sometimes projects can have same root - the current test case failed here. Workspace is used just to switch different configuration options for specific set of files. And there's also `coverage.allowExternal` 🤷

100-proof logic could be implemented if `coverage.include` was allowed in workspace projects. But I'd rather not expose any `coverage` options there.

Reply:
Yep, I'll add this.